src/testplan/testplanTestRunner.ts

```diff
@@ -0,0 +1,203 @@
+        public readonly adapterId: string,
+        private readonly logger: ILogger
+    ) { }
+
+    public cancel(): void {
+        this.testExecutions.forEach((execution, test) => {
+            this.logger.log('info', `Cancelling execution of ${test}`);
+            try {
+                execution.cancel();
+            } catch (error) {
+                this.logger.log('crit', `Cancelling execution of ${test} failed: ${error}`);
+            }
+        });
+    }
+
+    public async debugConfiguration(config: IWorkspaceConfiguration, test: string): Promise<IDebugConfiguration> {
+        const additionalEnvironment = await this.loadEnvironmentVariables(config);
+        const runArguments = this.getRunArguments(test, config.getTestplanConfiguration().testplanArguments);
+        return {
+            module: 'testplantest',
+            cwd: config.getCwd(),
+            args: runArguments.argumentsToPass,
+            env: additionalEnvironment,
+        };
+    }
+
+    public async load(config: IWorkspaceConfiguration): Promise<TestSuiteInfo | undefined> {
+        if (!config.getTestplanConfiguration().isTestplanEnabled) {
+            this.logger.log('info', 'TestPlan test discovery is disabled');
+            return undefined;
+        }
+        const additionalEnvironment = await this.loadEnvironmentVariables(config);
+        this.logger.log('info', `Discovering tests using python path '${config.pythonPath()}' in ${config.getCwd()}`);
+
+        const discoveryArguments = this.getDiscoveryArguments(config.getTestplanConfiguration().testplanArguments);
+        this.logger.log('info', `Running testplan with arguments: ${discoveryArguments.join(', ')}`);
+
+        const result = await this.runTestPlan(config, additionalEnvironment, discoveryArguments).complete();
+        const tests = parseTestSuites(result.output);
+        if (empty(tests)) {
+            this.logger.log('warn', 'No tests discovered');
+            return undefined;
+        }
+
+        setDescriptionForEqualLabels(tests, path.sep);
+        return {
+            type: 'suite',
+            id: this.adapterId,
+            label: 'Testplan tests',
+            children: tests,
+        };
+    }
+
+    public async run(config: IWorkspaceConfiguration, test: string): Promise<TestEvent[]> {
+        if (!config.getTestplanConfiguration().isTestplanEnabled) {
+            this.logger.log('info', 'Testplan test execution is disabled');
+            return [];
+        }
+        this.logger.log('info', `Running tests using python path '${config.pythonPath()}' in ${config.getCwd()}`);
+
+        const additionalEnvironment = await this.loadEnvironmentVariables(config);
+        const runArguments = this.getRunArguments(test, config.getTestplanConfiguration().testplanArguments);
+        const { dirName, cleanupCallback } = await this.getJunitReportPath(config.getCwd(), runArguments);
+        const testRunArguments = [
+            `--xml=${dirName}`
```

Comment:
I testplan docs I see it's possible to set `xml_dir` in tests - `@test_plan(name='SamplePlan', xml_dir='/path/to/xml-dir')`. What will happen in that case? Is `--xml` argument override this configuration?

Reply:
In the example it states that it overrides:
https://github.com/morganstanley/testplan/blob/0c9601b341be499a0781ae0bd935e1e9a7c69094/examples/Test%20Output/Exporters/XML/test_plan.py#L49